lightning/src/routing/network_graph.rs

```diff
@@ -831,22 +832,65 @@ pub enum EffectiveCapacity {
 	Infinite,
 	/// A capacity that is unknown possibly because either the chain state is unavailable to know
 	/// the total capacity or the `htlc_maximum_msat` was not advertised on the gossip network.
-	Unknown,
+	Unknown {
+		/// An amount which should be considered "already used". Used during routing to keep track
+		/// of how much we're already considering routing through this channel.
+		previously_used_msat: u64,
```

Comment:
Hmmm... I was hoping this would be a separate parameter to the scorer since it is relevant for all variants.

Reply:
> Sure, but its preferrable to tick something off the TODO list while we're at it rather than just fix the narrow case, no?

Not sure exactly how it would ultimately shake out, but I was thinking something along these lines: https://github.com/lightningdevkit/rust-lightning/pull/1227#discussion_r792802275. Had a rough start to this as part of some tabled work from #1227 (https://github.com/jkczyz/rust-lightning/commit/b88b37eabf35b2ee95f731571d1bc21679ddec48).

Then each scorer could use the in-use amount as they wish. `ProbabilisticScorer` would just need to subtract it from `max_liquidity_msat`.

I'd prefer if we make a small fix for 0.0.106 to avoid adding more review to be honest. Also, it somewhat conflicts with some work I pulled out of #1227 but haven't PR'ed: https://github.com/jkczyz/rust-lightning/commit/a9269138aef3a2a0ee98950e1887705309f71c36.

Reply:
> it somewhat conflicts with some work I pulled out of https://github.com/lightningdevkit/rust-lightning/pull/1227 but haven't PR'ed: https://github.com/jkczyz/rust-lightning/commit/a9269138aef3a2a0ee98950e1887705309f71c36.

Okay, sounds great! I'll close this and circle back around on the panic. I sent you the patch I'd started to build for this on discord, it looked like a performance win so probably good to incorporate it in your work.